var-core/src/test/java/io/varhttp/VarServletTest.java

```diff
@@ -0,0 +1,186 @@
+	}
+
+	@Test
+	public void handleGet_withRedirect_happyPath() throws IOException {
+		VarHttpServletRequest request = mock(VarHttpServletRequest.class);
+		when(request.getMethod()).thenReturn("GET");
+		when(request.getPathInfo()).thenReturn("/redirected-test");
+		when(request.getQueryString()).thenReturn("");
+		when(request.getContentType()).thenReturn("application/json");
+		when(request.getCharacterEncoding()).thenReturn("UTF-8");
+		when(request.getServletPath()).thenReturn("");
+		when(request.getContextPath()).thenReturn("");
+		when(request.getRequestURI()).thenReturn("/redirected-test");
+		when(request.getRequestURL()).thenReturn(new StringBuffer("/redirected-test"));
+		VarHttpServletResponse response = mock(VarHttpServletResponse.class);
+
+		ControllerExecution usedController = mock(ControllerExecution.class);
+		servlet.executions.put(context, new Request(HttpMethod.GET, "/test"), usedController);
+		ControllerExecution unusedController = mock(ControllerExecution.class);
+		servlet.executions.put(context, new Request(HttpMethod.GET, "/other"), unusedController);
+
+		when(response.getOutputStream()).thenReturn(mock(ServletOutputStream.class));
+
+		servlet.redirect("/redirected-test", "/test");
+
+		servlet.doGet(request, response);
+
+		verify(usedController, times(1)).execute(any());
+		verify(unusedController, never()).execute(any());
+		verify(response, times(0)).setStatus(404);
+	}
+
+	@Test
+	public void handleGet_withRedirect_toUnregisteredEndpoint() throws IOException {
+		VarHttpServletRequest request = mock(VarHttpServletRequest.class);
+		when(request.getMethod()).thenReturn("GET");
+		when(request.getPathInfo()).thenReturn("/redirected-test");
+		when(request.getQueryString()).thenReturn("");
+		when(request.getContentType()).thenReturn("application/json");
+		when(request.getCharacterEncoding()).thenReturn("UTF-8");
+		when(request.getServletPath()).thenReturn("");
+		when(request.getContextPath()).thenReturn("");
+		when(request.getRequestURI()).thenReturn("/redirected-test");
+		when(request.getRequestURL()).thenReturn(new StringBuffer("/redirected-test"));
+		VarHttpServletResponse response = mock(VarHttpServletResponse.class);
+
+		ControllerExecution usedController = mock(ControllerExecution.class);
+		servlet.executions.put(context, new Request(HttpMethod.GET, "/test"), usedController);
+		ControllerExecution unusedController = mock(ControllerExecution.class);
+		servlet.executions.put(context, new Request(HttpMethod.GET, "/other"), unusedController);
+
+		when(response.getOutputStream()).thenReturn(mock(ServletOutputStream.class));
+
+		servlet.redirect("/redirected-test", "/unregistered");
+
+		servlet.doGet(request, response);
+
+		verify(usedController, never()).execute(any());
+		verify(unusedController, never()).execute(any());
+		verify(response, times(1)).setStatus(404);
+	}
+
+	@Test
+	public void handleGet_withRedirectAndQueryString_queryStringIsIgnored() throws IOException {
```

Comment:
how do you check that it is ignored?

Reply:
By verifying that the correct endpoint controller is still called, even through the queryParams is set

Reply:
but if both have matching params, do they get set? ignored sounds like they should not, but that is not tested here

Reply:
95575a36